src/coreclr/src/utilcode/pedecoder.cpp

```diff
@@ -1770,20 +1770,31 @@ void PEDecoder::LayoutILOnly(void *base, BOOL allowFullPE) const
                            PAGE_READONLY, &oldProtection))
         ThrowLastError();
 
-    // Finally, apply proper protection to copied sections
-    section = sectionStart;
-    while (section < sectionEnd)
+    // Finally, apply proper protection to copied sections    
+    for (section = sectionStart; section < sectionEnd; section++)
     {
         // Add appropriate page protection.
-        if ((section->Characteristics & VAL32(IMAGE_SCN_MEM_WRITE)) == 0)
+#if defined(CROSSGEN_COMPILE) || defined(TARGET_UNIX)
+        if (section->Characteristics & IMAGE_SCN_MEM_WRITE)
```

Comment:
Is using C-style non-zero test a common practice in the CLR code base?

Reply:
It is fairly common, especially when it is a flag check or a part of a bool-normalizing operator ( !, &&, || ).

Sometimes comparing with `nullptr`, `NULL`, 0, `FALSE` adds clarity, sometimes generic "non-zero" is good enough and shorter. Containing file has a few examples of either case.